# Market Channel Analysis in Jan 2022  

Analyzed by Crystella Zheng

###Summary:
* Beginning on 1/12/2022, all free market channels and EDM were launched on a regular basis.
* Previously, we posted jobs and articles on 11 marketing channels each week. On 1/20/2022, there were 4 new channels discovered: 北美华人e网, 咕噜美国通, 一方论坛, 侠客行. However, 北美华人e网 was prohibited the day after it was released for unknown reasons. 一方论坛 & 侠客行 were introduced by Joyce.
* 其他includes paid market channels, forum reviews, unknown sources, slient inquirer and etc.

###Content:  
####1. Data Loading####
####2. Data Analyses & Visualization####
####3. Conclusion####

####1. Data Loading

In [0]:
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [0]:
pd = spark.sql('select * from channel_data').toPandas()
pd.display(5)

Date Day Channels Post
Daily Daily
Inquiry No Chat DailyLeadInquiry LeadInquiryRate Success
Inquiry TitleKeyWords 1/14/22 Fri 58同城 12.0 2 0 1 50.00% 0 DA全职, SDE全职, VA, AI/机器人公司 1/14/22 Fri Mitbbs 6.0 1 0 0 0.00% 0 DA全职, BA全职, Fintech, 夏季实习, FA, IA, DA实习全职, AI机器人公司, Methsoft 1/14/22 Fri bay123 6.0 5 0 3 60.00% 1 DA全职, BA全职, Fintech, 夏季实习, FA, IA, DA实习全职, AI机器人公司, Methsoft, 职场加速器, MA实习 1/14/22 Fri 华人工商 6.0 1 0 1 100.00% 0 DA全职, BA全职, Fintech, 夏季实习, FA, IA, VA, DA实习全职, VA, AI/机器人公司, Methsoft 1/14/22 Fri 西雅图社区 6.0 5 1 2 50.00% 0 DA全职, BA全职, Fintech, 夏季实习, FA, IA, VA, DA实习全职, VA, AI/机器人公司, Methsoft 1/14/22 Fri 北美微论坛 6.0 3 0 1 33.33% 0 DA全职, BA全职, Fintech, 夏季实习, FA, IA, VA, DA实习全职, VA, AI/机器人公司, 职场加速器, MA实习 1/14/22 Fri 留园网 6.0 0 0 0 #DIV/0! 0 DA全职, BA全职, Fintech, 夏季实习, FA, IA, VA, DA实习全职, VA, AI/机器人公司, Methsoft 1/14/22 Fri 金门汇 1.0 0 0 0 #DIV/0! 0 VA, DA全职, Ai机器人公司, SDE实习 1/14/22 Fri 1o24bbs 2.0 2 0 2 100.00% 1 VA, DA全职, Ai机器人公司, SDE实习 1/14/22 Fri Email 82.0 2 0 2 100.00% 0 Google, 面经, Leetcode, 按照, 顺序, 刷题, 比较好 1/14/22 Fri 其他 16.0 66 13 13 24.53% 0 硅谷知名机器人, SDE实习全职, DA实习全职, MetLife, 夏季实习, Fintech, SDE实习, CreditRiskAnalyst全职, 职场加速器, MA实习 1/17/22 Mon 58同城 0.0 0 0 0 #DIV/0! 0 
 1/17/22 Mon Mitbbs 0.0 2 0 2 100.00% 0 
 1/17/22 Mon bay123 0.0 0 0 0 #DIV/0! 0 
 1/17/22 Mon 华人工商 0.0 1 0 1 100.00% 0 
 1/17/22 Mon 西雅图社区 0.0 0 0 0 #DIV/0! 0 
 1/17/22 Mon 北美微论坛 0.0 0 0 0 #DIV/0! 0 
 1/17/22 Mon 留园网 0.0 0 0 0 #DIV/0! 0 
 1/17/22 Mon 金门汇 0.0 0 0 0 #DIV/0! 0 
 1/17/22 Mon 1o24bbs 0.0 1 0 1 100.00% 0 
 1/17/22 Mon Email 68.0 10 2 10 125.00% 0 
 1/17/22 Mon 其他 10.0 7 0 4 57.14% 1 
 1/18/22 Tue 58同城 6.0 0 0 0 #DIV/0! 0 SDE实习, Methsoft, 湾区大厂, SDE合同工 1/18/22 Tue Mitbbs 0.0 0 0 0 #DIV/0! 0 
 1/18/22 Tue bay123 0.0 1 0 0 0.00% 0 
 1/18/22 Tue 华人工商 0.0 0 0 0 #DIV/0! 0 
 1/18/22 Tue 西雅图社区 0.0 0 0 0 #DIV/0! 0 
 1/18/22 Tue 北美微论坛 0.0 1 0 0 0.00% 0 
 1/18/22 Tue 留园网 0.0 0 0 0 #DIV/0! 0 
 1/18/22 Tue 金门汇 0.0 0 0 0 #DIV/0! 0 
 1/18/22 Tue 1o24bbs 1.0 1 0 1 100.00% 0 深度回答, ML, 机器学习, 面试问题 1/18/22 Tue Email 0.0 0 0 0 #DIV/0! 0 
 1/18/22 Tue 其他 0.0 1 1 0 #DIV/0! 0 
 1/19/22 Wed 58同城 0.0 1 0 0 0.00% 0 
 1/19/22 Wed Mitbbs 1.0 0 0 0 #DIV/0! 0 湾区大厂, SDE合同工 1/19/22 Wed bay123 1.0 0 0 0 #DIV/0! 0 湾区大厂, SDE合同工 1/19/22 Wed 华人工商 1.0 0 0 0 #DIV/0! 0 湾区大厂, SDE合同工 1/19/22 Wed 西雅图社区 1.0 0 0 0 #DIV/0! 0 湾区大厂, SDE合同工 1/19/22 Wed 北美微论坛 1.0 0 0 0 #DIV/0! 0 湾区大厂, SDE合同工 1/19/22 Wed 留园网 1.0 0 0 0 #DIV/0! 0 湾区大厂, SDE合同工 1/19/22 Wed 金门汇 1.0 0 0 0 #DIV/0! 0 湾区大厂, SDE合同工 1/19/22 Wed 1o24bbs 0.0 0 0 0 #DIV/0! 0 
 1/19/22 Wed Email 0.0 2 0 2 100.00% 0 
 1/19/22 Wed 其他 0.0 2 0 1 50.00% 0 
 1/20/22 Thur 58同城 0.0 1 0 1 100.00% 0 
 1/20/22 Thur Mitbbs 0.0 0 0 0 #DIV/0! 0 
 1/20/22 Thur bay123 0.0 1 0 1 100.00% 0 
 1/20/22 Thur 华人工商 0.0 0 0 0 #DIV/0! 0 
 1/20/22 Thur 西雅图社区 0.0 0 0 0 #DIV/0! 0 
 1/20/22 Thur 北美微论坛 0.0 0 0 0 #DIV/0! 0 
 1/20/22 Thur 留园网 0.0 0 0 0 #DIV/0! 0 
 1/20/22 Thur 金门汇 0.0 0 0 0 #DIV/0! 0 
 1/20/22 Thur 北美华人e网 0.0 1 1 0 #DIV/0! 0 
 1/20/22 Thur 1o24bbs 0.0 0 0 0 #DIV/0! 0 
 1/20/22 Thur Email 0.0 1 0 1 100.00% 0 
 1/20/22 Thur 其他 8.0 2 0 2 100.00% 0 
 1/21/22 Fri 58同城 6.0 0 0 0 #DIV/0! 0 快手, HROperation, NewGrad 1/21/22 Fri Mitbbs 1.0 0 0 0 #DIV/0! 0 快手, HROperation, NewGrad 1/21/22 Fri bay123 1.0 0 0 0 #DIV/0! 0 快手, HROperation, NewGrad 1/21/22 Fri 华人工商 1.0 0 0 0 #DIV/0! 0 快手, HROperation, NewGrad 1/21/22 Fri 西雅图社区 1.0 0 0 0 #DIV/0! 0 快手, HROperation, NewGrad 1/21/22 Fri 北美微论坛 1.0 0 0 0 #DIV/0! 0 快手, HROperation, NewGrad 1/21/22 Fri 留园网 1.0 0 0 0 #DIV/0! 0 快手, HROperation, NewGrad 1/21/22 Fri 金门汇 1.0 0 0 0 #DIV/0! 0 快手, HROperation, NewGrad 1/21/22 Fri 1o24bbs null 0 0 0 #DIV/0! 0 
 1/21/22 Fri 北美华人e网 1.0 0 0 0 #DIV/0! 0 SDE全职, AI机器人公司 1/21/22 Fri 美国咕噜通 2.0 0 0 0 #DIV/0! 0 DA全职, SDE全职, VA, AI/机器人公司 1/21/22 Fri 一方APP 2.0 0 0 0 #DIV/0! 0 DA全职, SDE全职, VA, AI/机器人公司 1/21/22 Fri 侠客行 1.0 0 0 0 #DIV/0! 0 SDE全职, AI机器人公司 1/21/22 Fri Email null 0 0 0 #DIV/0! 0 
 1/21/22 Fri 其他 5.0 3 1 1 50.00% 0 
 1/24/22 Mon 58同城 0.0 0 0 0 #DIV/0! 0 
 1/24/22 Mon Mitbbs 0.0 1 0 0 0.00% 0 
 1/24/22 

In [0]:
pd.rename(columns={'Post\nDaily':'Post', 'Daily\nInquiry':'Inquiry', 'No Chat':'NoChat','DailyLeadInquiry':'Leads','Success\nInquiry':'Success', 'TitleKeyWords':'Title', 'Lead InquiryRate':'LeadInquiryRate'}, inplace=True)
pd.head()

Date 
 Day 
 Channels 
 Post 
 Inquiry 
 NoChat 
 Leads 
 LeadInquiryRate 
 Success 
 Title 
 
 
 
 
 0 
 1/14/22 
 Fri 
 58同城 
 12.0 
 2 
 0 
 1 
 50.00% 
 0 
 DA全职, SDE全职, VA, AI/机器人公司 
 
 
 1 
 1/14/22 
 Fri 
 Mitbbs 
 6.0 
 1 
 0 
 0 
 0.00% 
 0 
 DA全职, BA全职, Fintech, 夏季实习, FA, IA, DA实习全职, AI机... 
 
 
 2 
 1/14/22 
 Fri 
 bay123 
 6.0 
 5 
 0 
 3 
 60.00% 
 1 
 DA全职, BA全职, Fintech, 夏季实习, FA, IA, DA实习全职, AI机... 
 
 
 3 
 1/14/22 
 Fri 
 华人工商 
 6.0 
 1 
 0 
 1 
 100.00% 
 0 
 DA全职, BA全职, Fintech, 夏季实习, FA, IA, VA, DA实习全职,... 
 
 
 4 
 1/14/22 
 Fri 
 西雅图社区 
 6.0 
 5 
 1 
 2 
 50.00% 
 0 
 DA全职, BA全职, Fintech, 夏季实习, FA, IA, VA, DA实习全职,...

In [0]:
pd.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 141 entries, 0 to 140
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Date             141 non-null    object 
 1   Day              141 non-null    object 
 2   Channels         141 non-null    object 
 3   Post             139 non-null    float64
 4   Inquiry          141 non-null    int32  
 5   NoChat           141 non-null    int32  
 6   Leads            141 non-null    int32  
 7   LeadInquiryRate  141 non-null    object 
 8   Success          141 non-null    int32  
 9   Title            140 non-null    object 
dtypes: float64(1), int32(4), object(5)
memory usage: 8.9+ KB


In [0]:
# convert string date into datetime 
pd['Date']=pd['Date'].astype('datetime64[ns]')

# replace all non-numeric value in the Lead Ratio column into numeric value
pd = pd.replace('#DIV/0!','0.00%')

# convert percent string to float64
pd['LeadInquiryRate'] = pd['LeadInquiryRate'].str.rstrip('%').astype(float) / 100

In [0]:
print('Total days for Posting:', len(pd['Date'].unique()))
print('Total Channels:', len(pd['Channels'].unique()))
print('Total Posting and Emailing:', pd['Post'].sum())
print('Total Inquiry:', pd['Inquiry'].sum())
print('Total Leads:', pd['Leads'].sum())
print('Total NoChat:', pd['NoChat'].sum())
print('Total Success:', pd['Success'].sum())
print('Raito of Inquiry', round((pd['Leads'].sum()/pd['Inquiry'].sum())*100, 2),'%')

Total days for Posting: 11
Total Channels: 15
Total Posting and Emailing: 412.0
Total Inquiry: 166
Total Leads: 76
Total NoChat: 23
Total Success: 3
Raito of Inquiry 45.78 %


####2. Data Analyses & Visualization

In [0]:
df = spark.createDataFrame(pd)
print(df.printSchema)
df.createOrReplaceTempView("channel")

<bound method DataFrame.printSchema of DataFrame[Date: timestamp, Day: string, Channels: string, Post: double, Inquiry: int, NoChat: int, Leads: int, LeadInquiryRate: double, Success: int, Title: string]>


#####2.1 Timeline Analysis#####

In [0]:
%sql
select Date, sum(Post) as Daily_Total_Post, sum(Inquiry) as Daily_Total_Inquiry, sum(Leads) as Daily_Total_Leads, sum(Success) as Daily_Total_Success, sum(NoChat) as Daily_Total_NoChat
from channel
group by Date
order by Daily_Total_Leads desc, Daily_Total_Inquiry desc, Daily_Total_Post desc

Date Daily_Total_Post Daily_Total_Inquiry Daily_Total_Leads Daily_Total_Success Daily_Total_NoChat 2022-01-14T00:00:00.000+0000 149.0 87 25 2 14 2022-01-17T00:00:00.000+0000 78.0 21 18 1 2 2022-01-26T00:00:00.000+0000 9.0 17 9 0 1 2022-01-24T00:00:00.000+0000 66.0 11 8 0 2 2022-01-20T00:00:00.000+0000 8.0 6 5 0 1 2022-01-28T00:00:00.000+0000 11.0 6 3 0 1 2022-01-19T00:00:00.000+0000 7.0 5 3 0 0 2022-01-27T00:00:00.000+0000 45.0 4 2 0 0 2022-01-18T00:00:00.000+0000 7.0 4 1 0 1 2022-01-21T00:00:00.000+0000 24.0 3 1 0 1 2022-01-25T00:00:00.000+0000 8.0 2 1 0 0

In [0]:
%sql
select Date, sum(Post) as Daily_Total_Post, sum(Inquiry) as Daily_Total_Inquiry, sum(Leads) as Daily_Total_Leads, sum(Success) as Daily_Total_Success, sum(NoChat) as Daily_Total_NoChat
from channel
group by Date

Date Daily_Total_Post Daily_Total_Inquiry Daily_Total_Leads Daily_Total_Success Daily_Total_NoChat 2022-01-14T00:00:00.000+0000 149.0 87 25 2 14 2022-01-17T00:00:00.000+0000 78.0 21 18 1 2 2022-01-18T00:00:00.000+0000 7.0 4 1 0 1 2022-01-19T00:00:00.000+0000 7.0 5 3 0 0 2022-01-20T00:00:00.000+0000 8.0 6 5 0 1 2022-01-21T00:00:00.000+0000 24.0 3 1 0 1 2022-01-25T00:00:00.000+0000 8.0 2 1 0 0 2022-01-24T00:00:00.000+0000 66.0 11 8 0 2 2022-01-26T00:00:00.000+0000 9.0 17 9 0 1 2022-01-27T00:00:00.000+0000 45.0 4 2 0 0 2022-01-28T00:00:00.000+0000 11.0 6 3 0 1

In [0]:
%sql
select date_groupby.*, channel.Title
from
channel join
(select Date,sum(Post) as Daily_Total_Post, sum(Inquiry) as Daily_Total_Inquiry, sum(Leads) as Daily_Total_Leads, sum(Success) as Daily_Total_Success, sum(NoChat) as Daily_Total_NoChat
from channel
group by Date) as date_groupby on channel.Date=date_groupby.Date

Date Daily_Total_Post Daily_Total_Inquiry Daily_Total_Leads Daily_Total_Success Daily_Total_NoChat Title 2022-01-14T00:00:00.000+0000 149.0 87 25 2 14 DA全职, SDE全职, VA, AI/机器人公司 2022-01-14T00:00:00.000+0000 149.0 87 25 2 14 DA全职, BA全职, Fintech, 夏季实习, FA, IA, DA实习全职, AI机器人公司, Methsoft 2022-01-14T00:00:00.000+0000 149.0 87 25 2 14 DA全职, BA全职, Fintech, 夏季实习, FA, IA, DA实习全职, AI机器人公司, Methsoft, 职场加速器, MA实习 2022-01-14T00:00:00.000+0000 149.0 87 25 2 14 DA全职, BA全职, Fintech, 夏季实习, FA, IA, VA, DA实习全职, VA, AI/机器人公司, Methsoft 2022-01-14T00:00:00.000+0000 149.0 87 25 2 14 DA全职, BA全职, Fintech, 夏季实习, FA, IA, VA, DA实习全职, VA, AI/机器人公司, Methsoft 2022-01-14T00:00:00.000+0000 149.0 87 25 2 14 DA全职, BA全职, Fintech, 夏季实习, FA, IA, VA, DA实习全职, VA, AI/机器人公司, 职场加速器, MA实习 2022-01-14T00:00:00.000+0000 149.0 87 25 2 14 DA全职, BA全职, Fintech, 夏季实习, FA, IA, VA, DA实习全职, VA, AI/机器人公司, Methsoft 2022-01-14T00:00:00.000+0000 149.0 87 25 2 14 VA, DA全职, Ai机器人公司, SDE实习 2022-01-14T00:00:00.000+0000 149.0 87 25 2 14 VA, DA全职, Ai机器人公司, SDE实习 2022-01-14T00:00:00.000+0000 149.0 87 25 2 14 Google, 面经, Leetcode, 按照, 顺序, 刷题, 比较好 2022-01-14T00:00:00.000+0000 149.0 87 25 2 14 硅谷知名机器人, SDE实习全职, DA实习全职, MetLife, 夏季实习, Fintech, SDE实习, CreditRiskAnalyst全职, 职场加速器, MA实习 2022-01-17T00:00:00.000+0000 78.0 21 18 1 2 
 2022-01-17T00:00:00.000+0000 78.0 21 18 1 2 
 2022-01-17T00:00:00.000+0000 78.0 21 18 1 2 
 2022-01-17T00:00:00.000+0000 78.0 21 18 1 2 
 2022-01-17T00:00:00.000+0000 78.0 21 18 1 2 
 2022-01-17T00:00:00.000+0000 78.0 21 18 1 2 
 2022-01-18T00:00:00.000+0000 7.0 4 1 0 1 SDE实习, Methsoft, 湾区大厂, SDE合同工 2022-01-18T00:00:00.000+0000 7.0 4 1 0 1 
 2022-01-18T00:00:00.000+0000 7.0 4 1 0 1 
 2022-01-18T00:00:00.000+0000 7.0 4 1 0 1 
 2022-01-18T00:00:00.000+0000 7.0 4 1 0 1 
 2022-01-18T00:00:00.000+0000 7.0 4 1 0 1 
 2022-01-18T00:00:00.000+0000 7.0 4 1 0 1 
 2022-01-18T00:00:00.000+0000 7.0 4 1 0 1 
 2022-01-18T00:00:00.000+0000 7.0 4 1 0 1 深度回答, ML, 机器学习, 面试问题 2022-01-18T00:00:00.000+0000 7.0 4 1 0 1 
 2022-01-18T00:00:00.000+0000 7.0 4 1 0 1 
 2022-01-19T00:00:00.000+0000 7.0 5 3 0 0 
 2022-01-19T00:00:00.000+0000 7.0 5 3 0 0 湾区大厂, SDE合同工 2022-01-17T00:00:00.000+0000 78.0 21 18 1 2 
 2022-01-17T00:00:00.000+0000 78.0 21 18 1 2 
 2022-01-17T00:00:00.000+0000 78.0 21 18 1 2 
 2022-01-17T00:00:00.000+0000 78.0 21 18 1 2 
 2022-01-17T00:00:00.000+0000 78.0 21 18 1 2 
 2022-01-20T00:00:00.000+0000 8.0 6 5 0 1 
 2022-01-20T00:00:00.000+0000 8.0 6 5 0 1 
 2022-01-20T00:00:00.000+0000 8.0 6 5 0 1 
 2022-01-20T00:00:00.000+0000 8.0 6 5 0 1 
 2022-01-20T00:00:00.000+0000 8.0 6 5 0 1 
 2022-01-20T00:00:00.000+0000 8.0 6 5 0 1 
 2022-01-20T00:00:00.000+0000 8.0 6 5 0 1 
 2022-01-20T00:00:00.000+0000 8.0 6 5 0 1 
 2022-01-19T00:00:00.000+0000 7.0 5 3 0 0 湾区大厂, SDE合同工 2022-01-19T00:00:00.000+0000 7.0 5 3 0 0 湾区大厂, SDE合同工 2022-01-19T00:00:00.000+0000 7.0 5 3 0 0 湾区大厂, SDE合同工 2022-01-19T00:00:00.000+0000 7.0 5 3 0 0 湾区大厂, SDE合同工 2022-01-19T00:00:00.000+0000 7.0 5 3 0 0 湾区大厂, SDE合同工 2022-01-19T00:00:00.000+0000 7.0 5 3 0 0 湾区大厂, SDE合同工 2022-01-19T00:00:00.000+0000 7.0 5 3 0 0 
 2022-01-19T00:00:00.000+0000 7.0 5 3 0 0 
 2022-01-19T00:00:00.000+0000 7.0 5 3 0 0 
 2022-01-21T00:00:00.000+0000 24.0 3 1 0 1 快手, HROperation, NewGrad 2022-01-21T00:00:00.000+0000 24.0 3 1 0 1 快手, HROperation, NewGrad 2022-01-21T00:00:00.000+0000 24.0 3 1 0 1 快手, HROperation, NewGrad 2022-01-21T00:00:00.000+0000 24.0 3 1 0 1 快手, HROperation, NewGrad 2022-01-21T00:00:00.000+0000 24.0 3 1 0 1 快手, HROperation, NewGrad 2022-01-21T00:00:00.000+0000 24.0 3 1 0 1 快手, HROperation, NewGrad 2022-01-21T00:00:00.000+0000 24.0 3 1 0 1 快手, HROperation, NewGrad 2022-01-21T00:00:00.000+0000 24.0 3 1 0 1 快手, HROperation, NewGrad 2022-01-21T00:00:00.000+0000 24.0 3 1 0 1 
 2022-01-21T00:00:00.000+0000 24.0 3 1 0 1 SDE全职, AI机器人公司 2022-01-21T00:00:00.000+0000 24.0 3 1 0 1 DA全职, SDE全职, VA, AI/机器人公司 2022-01-21T00:00:00.000+0000 24.0 3 1 0 1 DA全职, SDE全职, VA, AI/机器人公司 2022-01-21T00:00:00.000+0000 24.0 3 1 0 1 SDE全职, AI机器人公司 2022-01-21T00:00:

1. The number of Daily Total Post on Jan 14 looks at the peak on the trend but actually the result was accumulated from Jan 1 to Jan 14. Other than that, the peak of Daily Total Post should be at 125 on Jan 17 instead, which was the approaches of EDM, posting and articles. Another 2 peaks were on Jan 24 Monday, followed by Jan 27 Thursday.
2. Similary to Daily Total Post records, the peak of Daily Inquiry also fell at Jan 17, which was 21 inquiries. Then the lower crest arrived at Jan 26 Wednesday, of which WeChat Inquier was 17 and the reason that was the vacancy post of 'BA实习 at Vobile'.
3. The trend of Daily Total Leads imitated Daily Total Inquiry's. On the other hand, the total of Success had no relevance to the current acts.

#####2.2 Day Analysis#####

In [0]:
%sql
select Day, sum(Post) as Daily_Total_Post, sum(Inquiry) as Daily_Total_Inquiry, sum(Leads) as Daily_Total_Leads, sum(Success) as Daily_Total_Success, sum(NoChat) as Daily_Total_NoChat
from channel
group by Day

Day Daily_Total_Post Daily_Total_Inquiry Daily_Total_Leads Daily_Total_Success Daily_Total_NoChat Mon 144.0 32 26 1 4 Fri 184.0 96 29 2 16 Wed 16.0 22 12 0 1 Tue 15.0 6 2 0 1 Thur 53.0 10 7 0 1

1. Under the normal circumstances, people prefer looking for jobs on Monday and browsing Internet on Friday, followed by Thursday. Therefore, posting act is aligned with these circumstances. 
2. Obviously, the majority of Inquiry and Leads occured on Friday, followed by Monday.

#####2.3 Channel Analysis#####

In [0]:
%sql
select Channels, sum(Post) as Daily_Total_Post, sum(Inquiry) as Daily_Total_Inquiry, sum(Leads) as Daily_Total_Leads, sum(Success) as Daily_Total_Success, sum(NoChat) as Daily_Total_NoChat
from channel
group by Channels

Channels Daily_Total_Post Daily_Total_Inquiry Daily_Total_Leads Daily_Total_Success Daily_Total_NoChat 1o24bbs 4.0 5 5 1 0 华人工商 10.0 2 2 0 0 bay123 10.0 9 4 1 0 北美微论坛 10.0 5 1 0 1 Email 268.0 17 17 0 2 留园网 10.0 0 0 0 0 58同城 24.0 5 2 0 0 Mitbbs 10.0 4 2 0 0 西雅图社区 10.0 20 12 0 1 金门汇 5.0 0 0 0 0 其他 39.0 92 26 1 17 美国咕噜通 7.0 2 2 0 0 北美华人e网 1.0 1 0 0 1 一方APP 2.0 4 3 0 1 侠客行 2.0 0 0 0 0

In [0]:
%sql
select Channels, sum(Post) as Daily_Total_Post, sum(Inquiry) as Daily_Total_Inquiry, sum(Leads) as Daily_Total_Leads, sum(Success) as Daily_Total_Success, sum(NoChat) as Daily_Total_NoChat
from channel
group by Channels

Channels Daily_Total_Post Daily_Total_Inquiry Daily_Total_Leads Daily_Total_Success Daily_Total_NoChat 1o24bbs 4.0 5 5 1 0 华人工商 10.0 2 2 0 0 bay123 10.0 9 4 1 0 北美微论坛 10.0 5 1 0 1 Email 268.0 17 17 0 2 留园网 10.0 0 0 0 0 58同城 24.0 5 2 0 0 Mitbbs 10.0 4 2 0 0 西雅图社区 10.0 20 12 0 1 金门汇 5.0 0 0 0 0 其他 39.0 92 26 1 17 美国咕噜通 7.0 2 2 0 0 北美华人e网 1.0 1 0 0 1 一方APP 2.0 4 3 0 1 侠客行 2.0 0 0 0 0

#####2.4 Leads Ratio Analysis#####

In [0]:
%sql
select Date, Day, Channels, LeadInquiryRate, Title
from channel
where LeadInquiryRate > 0.6 and Inquiry >= 2

Date Day Channels LeadInquiryRate Title 2022-01-14T00:00:00.000+0000 Fri 1o24bbs 1.0 VA, DA全职, Ai机器人公司, SDE实习 2022-01-14T00:00:00.000+0000 Fri Email 1.0 Google, 面经, Leetcode, 按照, 顺序, 刷题, 比较好 2022-01-17T00:00:00.000+0000 Mon Mitbbs 1.0 
 2022-01-17T00:00:00.000+0000 Mon Email 1.25 
 2022-01-19T00:00:00.000+0000 Wed Email 1.0 
 2022-01-20T00:00:00.000+0000 Thur 其他 1.0 
 2022-01-24T00:00:00.000+0000 Mon 一方APP 1.0 
 2022-01-24T00:00:00.000+0000 Mon Email 1.0 
 2022-01-24T00:00:00.000+0000 Mon 其他 1.0 
 2022-01-26T00:00:00.000+0000 Wed 西雅图社区 0.6667000000000001 Vobile, BA实习 2022-01-27T00:00:00.000+0000 Thur 西雅图社区 1.0 
 2022-01-28T00:00:00.000+0000 Fri 美国咕噜通 1.0 一站式, 金融, 美国商业银行, 金融类, BD实习, FA实习, IA实习, Compliance实习

In [0]:
%sql
select Date, Day, Channels, LeadInquiryRate, Title
from channel
where LeadInquiryRate > 0.6 and Inquiry >= 2

Date Day Channels LeadInquiryRate Title 2022-01-14T00:00:00.000+0000 Fri 1o24bbs 1.0 VA, DA全职, Ai机器人公司, SDE实习 2022-01-14T00:00:00.000+0000 Fri Email 1.0 Google, 面经, Leetcode, 按照, 顺序, 刷题, 比较好 2022-01-17T00:00:00.000+0000 Mon Mitbbs 1.0 
 2022-01-17T00:00:00.000+0000 Mon Email 1.25 
 2022-01-19T00:00:00.000+0000 Wed Email 1.0 
 2022-01-20T00:00:00.000+0000 Thur 其他 1.0 
 2022-01-24T00:00:00.000+0000 Mon 一方APP 1.0 
 2022-01-24T00:00:00.000+0000 Mon Email 1.0 
 2022-01-24T00:00:00.000+0000 Mon 其他 1.0 
 2022-01-26T00:00:00.000+0000 Wed 西雅图社区 0.6667000000000001 Vobile, BA实习 2022-01-27T00:00:00.000+0000 Thur 西雅图社区 1.0 
 2022-01-28T00:00:00.000+0000 Fri 美国咕噜通 1.0 一站式, 金融, 美国商业银行, 金融类, BD实习, FA实习, IA实习, Compliance实习

1. Due of the latency and instability difficulties associated with EDM and 其他, they will be discarded. However, it was still observed that people tend to drop emails on Monday, Wednesday and Friday. Perhaps, 
2. 1o24bbs, 咕噜美国通, 一方论坛, Mitbbs and especially 西雅图社区 outperformed slight above average.
3. Friday: 1o24bbs，咕噜美国通; Monday: Mitbbs & 一方论坛; Thur: 西雅图社区.
4. Watch out! No performance occurred on Tuesday at all!

#####2.5 Tokens Analysis#####

In [0]:
max_title_df = spark.sql('select Date, Day, Channels, LeadInquiryRate, Title from channel where LeadInquiryRate > 0.6 and Inquiry >= 2')

max_title_df.display()

Date Day Channels LeadInquiryRate Title 2022-01-14T00:00:00.000+0000 Fri 1o24bbs 1.0 VA, DA全职, Ai机器人公司, SDE实习 2022-01-14T00:00:00.000+0000 Fri Email 1.0 Google, 面经, Leetcode, 按照, 顺序, 刷题, 比较好 2022-01-17T00:00:00.000+0000 Mon Mitbbs 1.0 
 2022-01-17T00:00:00.000+0000 Mon Email 1.25 
 2022-01-19T00:00:00.000+0000 Wed Email 1.0 
 2022-01-20T00:00:00.000+0000 Thur 其他 1.0 
 2022-01-24T00:00:00.000+0000 Mon 一方APP 1.0 
 2022-01-24T00:00:00.000+0000 Mon Email 1.0 
 2022-01-24T00:00:00.000+0000 Mon 其他 1.0 
 2022-01-26T00:00:00.000+0000 Wed 西雅图社区 0.6667000000000001 Vobile, BA实习 2022-01-27T00:00:00.000+0000 Thur 西雅图社区 1.0 
 2022-01-28T00:00:00.000+0000 Fri 美国咕噜通 1.0 一站式, 金融, 美国商业银行, 金融类, BD实习, FA实习, IA实习, Compliance实习

In [0]:
max_title_pd = max_title_df.toPandas()
max_title_pd.head()

Date 
 Day 
 Channels 
 LeadInquiryRate 
 Title 
 
 
 
 
 0 
 2022-01-14 
 Fri 
 1o24bbs 
 1.00 
 VA, DA全职, Ai机器人公司, SDE实习 
 
 
 1 
 2022-01-14 
 Fri 
 Email 
 1.00 
 Google, 面经, Leetcode, 按照, 顺序, 刷题, 比较好 
 
 
 2 
 2022-01-17 
 Mon 
 Mitbbs 
 1.00 
 \r 
 
 
 3 
 2022-01-17 
 Mon 
 Email 
 1.25 
 \r 
 
 
 4 
 2022-01-19 
 Wed 
 Email 
 1.00 
 \r

In [0]:
title_lst = max_title_pd.loc[:, 'Title'].tolist()
title_lst

Out[39]: ['VA, DA全职, Ai机器人公司, SDE实习',
 'Google, 面经, Leetcode, 按照, 顺序, 刷题, 比较好',
 '\r',
 '\r',
 '\r',
 '\r',
 '\r',
 '\r',
 '\r',
 'Vobile, BA实习',
 '\r',
 '一站式, 金融, 美国商业银行, 金融类, BD实习, FA实习, IA实习, Compliance实习']

In [0]:
def tokenization(text):
  tokens = []
  for token in text:
    if token == '\r':
      del token
    elif token is None:
      continue
    else:
      tokens.extend(token.split(','))
  return tokens

dup_tokens = tokenization(title_lst)
res=[]
for i in dup_tokens:
  if i not in res:
    res.append(i)

print("These are the keywords in the titles that drove 60% ratio: " + str(res))

These are the keywords in the titles that drove 60% ratio: ['VA', ' DA全职', ' Ai机器人公司', ' SDE实习', 'Google', ' 面经', ' Leetcode', ' 按照', ' 顺序', ' 刷题', ' 比较好', 'Vobile', ' BA实习', '一站式', ' 金融', ' 美国商业银行', ' 金融类', ' BD实习', ' FA实习', ' IA实习', ' Compliance实习']


According from the limit data in Jan, the lead ratio would be and beyond 60% if these the vacancy titles contain the following tokens:
> 'VA', ' DA全职', ' Ai机器人公司', ' SDE实习', 'Google', ' 面经', ' Leetcode', ' 按照', ' 顺序', ' 刷题', ' 比较好', 'Vobile', ' BA实习', '一站式', ' 金融', ' 美国商业银行', ' 金融类', ' BD实习', ' FA实习', ' IA实习', ' Compliance实习'  

(PS: VA='生物制药公司')

###3. Conclusion###

* The 'BA实习 at Vobile' on 1/26/2022 Wed became trending all of a sudden, which drew 17 inquirers to add on WeChat and 9 out of 17 belonged to Leads. From the perspective of the new grad and other young jobseekers, the fancy "Company About" somehow outweighted. The company description should be embellished. Nevertheless, internship can never be neglected.

* Data storytold us of the majority inquiries happening on Monday and Friday, which verified that my prematured operation on all these channels was correct.

* There was hesitation for hectic job posting due to account banning, etc. 美国华人e网. Now the job posting is 3 times per week; articles posting is 1-2 times per week; EDM is 1.5 time per week due to time-consuming; forum commenting is 1 time per week.

* Targets in Feb: total leads should deliver at 100 according to the monthly success goal(3) and proposed success ratio at 3%; there will be 217 inquiries according to lead ratio at 45.78% this month; therefore, it's required to have 7 inquiries on a daily basis.